backend/src/transactions/user.go

```diff
@@ -16,3 +17,16 @@ func GetAllUsers(db *gorm.DB) ([]models.User, error) {
 
 	return users, nil
 }
+
+func GetUser(db *gorm.DB, id string) (*models.User, error) {
+	var user models.User
+	if err := db.First(&user, id).Error; err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			return nil, fiber.NewError(fiber.StatusNotFound, err.Error())
```

Comment:
let's add a custom message here instead of spitting back the full error. down the line we could log the complete error but i would be hesitant returning the full `err.Error()` to the caller

Reply:
This depends on the caller right? If this was a public facing API i'd agree but we are the callers in this case.

Reply:
you're right that this is private but it will make it easier on the frontend if we have a good custom error message here, I'll let the frontend goat @alderwhiteford chime in here

Reply:
Also I don't see a way to predict what the error could be, so hard coding something is bad design...

Reply:
done